src/lib/codegen.ml

```diff
@@ -46,15 +46,21 @@ module Interface = struct
          BoxedFunction Method.({ name = m.name; description = m.description; ty = prepend m.ty}))
          interface.methods}
 
+  let rec setify = function
+    | [] -> []
+    | (x::xs) -> if List.mem x xs then setify xs else x::(setify xs)
```

Comment:
You could use a fold and make it tail recursive to avoid risk of stackoverflows (like the ones we had with `List.map`), but I don't think there will ever have that many errors